### PR TITLE
Headings and forms accessibility improvements

### PR DESCRIPTION
**Pull request by shyusu4**
This PR addresses accessibility issues related to heading levels and forms:

- Fixed instances (in hero, headline, and icon-link blocks) where headings were used consecutively after one another.
- Changed menu_get_started to use an `h2`
- #457
- #459
- Currently screen reader announces sign up forms on the site with "Enter your email address" repeated 3 times. Fixed by changing base_form to use `aria-label="Email"` instead of `<label>`
- Changes the use of `fieldset` in search-form to a `div` (fieldset is used to group multiple form fields together, but there's only one here)

<details>
  <summary>Screenshots</summary>

  | before | after |
  | --- | --- |
  |Screenshot (109)|Screenshot (105)|
Screenshot (110)|Screenshot (106)|

</details>

### CODE REVIEW

**Comment by Scotchester** [comment]
> Fixed instances (in hero, headline, and icon-link blocks) where headings were used consecutively after one another.

I looked for an issue in the audit spreadsheet that this addresses but couldn't easily find one. Could you point it out for me?

I'm a bit concerned about the addition of `hgroup` elements. They're probably harmless in the short term, but [according to MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hgroup), they have been removed from the HTML spec and are discouraged from being used. If a wrapper is helpful for applying the `role="group" aria-roledescription="heading group"` attributes, a `div` is probably best in this case.

File: wagtailio/static/sass/components/_sign-up-form.scss [hunk 1]
```diff
@@ -25,19 +25,22 @@
...
         }
 
         > svg {
             position: relative;
-            inset: -100px 0 auto auto;
             width: 170px;
             height: 170px;
+
+            @include media-query(medium) {
+                inset: -100px 0 auto auto;
```
Comment (Morsey187) [inline wagtailio/static/sass/components/_sign-up-form.scss:42]: Nickpick: This is a huge improvement over the current styling, however it's still possble to have the text overlap at some viewports, which given the bright colour of the icon here makes the text really difficuly to read.

<details><summary>Example</summary>

Screenshot 2024-02-06 at 17 39 07

</details>

If you altered the DOM a little, you could then constrain the text to a division element and have both the text and icon be responsive to their respective containers, using `display: flex`.


<details><summary>Example using flex</summary>
You could then use `flex-direction: column` to move the icon below the form at mobile. We'd also have to wouldn't have to offset the image using CSS which can be tricky to manage. 

Screenshot 2024-02-06 at 17 44 03

</details>

**Review by Morsey187 — COMMENTED** [review]
This looks great, I haven't looked so much into the use of aria tags and  hgroup as I noticed Scot's comment and certain things were discussed already on Zoom so I think Thibaud is better placed to review that aspect. 

Instead I've checked out the branch and tested locally, the SCSS looks fine to me I've just left some nickpicks/suggestions around text overlapping.

File: wagtailio/project_styleguide/templates/patterns/includes/search-form.html [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <form action="{% url 'search' %}" method="get" role="search" class="search {% if classes %}{{ classes }}{% endif %}" {% if data_attrs %}{{ data_attrs }}{% endif %} {% if form_id %}id="{{ form_id }}"{% endif %}>
-    <fieldset class="search__wrap">
+    <div class="search__wrap">
```
Comment (thibaudcolas) [inline wagtailio/project_styleguide/templates/patterns/includes/search-form.html:2]: Looking good!

**Review by thibaudcolas — COMMENTED** [review]
There are a few things that need changing, and I think more work needed on the text overlap, but this goes in the right direction.

**Review by thibaudcolas — APPROVED** [review]
Merging without the `role` and `aria-roledescription` usage!

**Pull request merged**